### PR TITLE
Implement -debugDescription for Request and Response

### PR DESCRIPTION
**Pull request by iOSCowboy**
Implement `-debugDescription` methods for `SPTDataLoaderRequest` and `SPTDataLoaderResponse`. 

Format of the output is the same as `NSURLResponse` and `NSURLRequest`.

### CODE REVIEW

**Comment by iOSCowboy** [comment]
Review: @rastersize, @8W9aG, @jgavris

**Comment by 8W9aG** [comment]
You'll need to add tests for this (in order to keep coverage up)

File: SPTDataLoader/SPTDataLoaderResponse.m [hunk 1]
```diff
@@ -181,4 +181,10 @@ - (NSDate *)retryAfterForHeaders:(NSDictionary *)headers
     return [httpDateFormatter dateFromString:retryAfterValue];
 }
 
+- (NSString *)debugDescription
+{
+    return [NSString stringWithFormat:@"%@ { URL: %@ } { status code: %lu, headers: %@ }", [super debugDescription], self.response.URL, (long)self.statusCode, self.responseHeaders];
```
Comment (rastersize) [inline SPTDataLoader/SPTDataLoaderResponse.m]: You've specified that the status code should be `long unsigned` but you then cast it to only `long`.

**Comment by rastersize** [comment]
With some tests and the type fix this will be golden :smiley:

**Comment by iOSCowboy** [comment]
Good catch, @rastersize. Changed them to `%ld` (`SPTDataLoaderResponseHTTPStatusCode` is signed). Added unit tests.

**Comment by rastersize** [comment]
:+1:

**Comment by iOSCowboy** [comment]
hooray! :tada:

**Comment by jgavris** [comment]
Can also use `%tu` for unsigned and and `%zd` for signed.

**Pull request merged**